shared/IConsole.cs

```diff
@@ -15,5 +15,7 @@ public interface IConsole
         bool IsInputRedirected { get; }
         bool IsOutputRedirected { get; }
         bool IsErrorRedirected { get; }
+        ConsoleColor ForegroundColor { get; set; }
```

Comment:
Hmm I don't recall exactly, but I think at one point we said we wanted patches to be 100% binary compatible, which means that no "public" breaking change can happen at all, even in `.Internal` namespaces. Is there a way to make this change without breaking any API surface?